runsc/specutils/specutils.go

```diff
@@ -758,3 +758,12 @@ func ResolveEnvs(envs ...[]string) ([]string, error) {
 func FaqErrorMsg(anchor, msg string) string {
 	return fmt.Sprintf("%s; see https://gvisor.dev/faq#%s for more details", msg, anchor)
 }
+
+// GetDefEnv gets default environment variables for running sentry.
+// The environment is set when creating the process and re-run the
+// process via call self, currently include gofer process and
+// sandbox/boot process.
+func GetDefEnv() []string {
+	env := []string{"MALLOC_ARENA_MAX=64", "GODEBUG=cgocheck=0", "GOMAXPROCS=16", "GOTRACEBACK=crash"}
```

Comment:
We can't force these settings on all users. Ideally these aren't necessary and we can just delete them.

But if they're important then we need some other way of setting them, for example:

1. Have `StartInNS` take a slice of environment variables and add a runsc flag to set that list. Something like `--ns-env=MALLOC_ARENA_MAX=64 --ns-env=GODEBUG=cgocheck=0` might work. This is probably ideal as it's most flexible.
1. Add a conditionally compiled file that contains these settings and include it (via Go build tags) only in the `//runsc:runsc-plugin-stack` target.
1. Maintain this part as a patch.

I think the first option is the best.

Reply:
Yes, I agree that these settings are not necessary for all users, but if we want to use plugin stack as network type, which imports cgo, it will need to set "GODEBUG=cgocheck=0". What I came up with is whether we could as an extra check in createSandboxProcess, if conf.Network is set to plugin stack, then we append this  "GODEBUG=cgocheck=0" to cmd.Env which will be used later in StartInNS. In this way, we could set this environment variable only when plugin stack is used.